src/main/java/seedu/apollo/storage/Storage.java

```diff
@@ -251,6 +272,27 @@ private static ModuleList readModuleFileContents(File save, Ui ui, ModuleList al
         return newModuleList;
     }
 
+    private static void addLessons(Module module, Module searchModule, String[] moduleInfo) {
+        module.createNewTimeTable();
+
+        for (int i = 1; i < moduleInfo.length; i++) {
+            String[] lessonInfo = moduleInfo[i].split(":");
+
+            for (Timetable timetable: searchModule.getModuleTimetable()) {
+                if (timetable.getLessonType().equals(lessonInfo[0])
+                        && timetable.getClassnumber().equals(lessonInfo[1])) {
+
+                    if (!module.getModuleTimetable().contains(timetable)) {
+                        module.getModuleTimetable().add(timetable);
+                    }
+
+                }
+            }
+        }
```

Comment:
Nesting is a bit deep here, can consider using Boolean values to replace the if statements? Would make the code more readable too, since it's not explicitly clear what the statements are checking for now.